services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/schema/EntitySchemaValidator.java

```diff
@@ -154,11 +154,12 @@ boolean setDerivedAnnotationsAndBindAccessRequirements(RestrictableObjectDescrip
 			Set<Long> accessRequirmentIdsToBind) {
 		accessRequirementManager.setDynamicallyBoundAccessRequirementsForSubject(objectDescriptor, accessRequirmentIdsToBind);
 		if(annotations != null) {
+			if (annotations.equals(derivedAnnotationDao.getDerivedAnnotations(objectDescriptor.getId()).orElse(null))) {
```

Comment:
Let us extract the "existingDerivedAnnotations" to its own variable before the comparison